### PR TITLE
Migrations-1223 - Improve Documentation

### PR DESCRIPTION
**Pull request by okhasawn**
### Description
This PR enhances the current documentation/readme files in the repo.

### Issues Resolved
[Migrations-1223](https://opensearch.atlassian.net/browse/MIGRATIONS-1223)

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,61 +6,52 @@ On wings of the morning breeze
...
+    - [Local Docker Solution](#local-docker-solution)
+    - [End-to-End Testing](#end-to-end-testing)
+    - [Cloud Deployment](#deploying-to-aws-with-copilot)
+    - [Security](#security)
+    - [License](#license)
 
 ## Setup for commits
 
-Developers must run the "install_githooks.sh" script in order to add the pre-commit hook.
+Developers must run the "install_githooks.sh" script in order to add the pre-commit hook, which runs a linting command
```
Comment (sumobrian) [inline README.md:25]: Change "the pre-commit hook" to "any pre-commit hooks"

File: README.md [hunk 2]
```diff
@@ -6,61 +6,52 @@ On wings of the morning breeze
...
+on the `*.py` files.
 
-## Docker Solution
+## Traffic Capture
 
 The TrafficCapture directory hosts a set of projects designed to facilitate the proxying and capturing of HTTP
```
Comment (sumobrian) [inline README.md:30]: Replace HTTP with HTTP(S)

File: README.md [hunk 3]
```diff
@@ -6,61 +6,52 @@ On wings of the morning breeze
 Effortless migration
 ```
 
-This repo will contain code and documentation to assist in migrations and upgrades of OpenSearch clusters.
+This repo will contain code, tools and documentation to assist in migrations to and upgrades of OpenSearch clusters.
```
Comment (sumobrian) [inline README.md:9]: Change "This repo will contain code" to "This repo contains code"

File: TrafficCapture/dockerSolution/README.md [hunk 4]
```diff
@@ -16,4 +16,33 @@ it will be mounted to the traffic-comparator and jupyter services.
...
+
+Must have Java version 11 installed.
+
+The tools in this directory can only be built if you have Java version 11 installed.
+
+The version is specified in `TrafficCapture/build.gradle` using a Java toolchain, which allows us
+to decouple the Java version used by Gradle itself from Java version used by the tools here.
+
+Any attempt to use a different version will cause the build to fail and will result in the following error (or similar)
+depending on which tool/project is being built. The below example shows the error printed when running e.g `./gradlew
```
Comment (sumobrian) [inline TrafficCapture/dockerSolution/README.md:37]: "The below example shows the error printed when running"

-->
Below is an example error when attempting to build with an incompatible Java version.

File: TrafficCapture/dockerSolution/README.md [hunk 5]
```diff
@@ -16,4 +16,33 @@ it will be mounted to the traffic-comparator and jupyter services.
...
+
+### Running the Docker Solution
+
+While in the TrafficCapture directory, run the following command:
+
+`./gradlew :dockerSolution:composeUp`
+
+### Compatibility
+
+Must have Java version 11 installed.
```
Comment (sumobrian) [inline TrafficCapture/dockerSolution/README.md:29]: I would remove this sentence and leave the one below.

File: test/README.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+## End-to-End Testing
+
+Developers can run a test script which will verify the end-to-end Docker Solution.
+#### Pre-requisites
```
Comment (sumobrian) [inline test/README.md:4]: Spacing?

File: test/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+* Have all containers from Docker solution running.
+
+To run the test script, users must navigate to this directory,
+install the required packages and then run the script:
+
+```
+cd test
+pip install -r requirements.txt
```
Comment (sumobrian) [inline test/README.md:13]: Python version should also be called out above under compatability.

**Review by sumobrian — CHANGES_REQUESTED** [review]
Maybe I'm missing it but I don't see documentation on "running the tools." I would add information about the OpenSearch benchmark container, provide an example run, and reference the OpenSearch benchmark documentation. 

Documenting how data flows from OpenSearch benchmarks to the replayer would be ideal. Keep in mind that the replayer responses are going to be stored in persistent storage. I would reference this but leave further documentation about the Comparator out. 

Can you open another task specifically for documenting how to analyze output? Also another task for cleanup if it doesn't exist?

**Comment by okhasawn** [comment]
> Maybe I'm missing it but I don't see documentation on "running the tools." I would add information about the OpenSearch benchmark container, provide an example run, and reference the OpenSearch benchmark documentation.
> 
> Documenting how data flows from OpenSearch benchmarks to the replayer would be ideal. Keep in mind that the replayer responses are going to be stored in persistent storage. I would reference this but leave further documentation about the Comparator out.
> 
> Can you open another task specifically for documenting how to analyze output? Also another task for cleanup if it doesn't exist?

@sumobrian, What I had in mind around "running the tools" was the already existing documentation on how to stand up the Docker Solution and/or the current AWS Deployment. But documentation around the OpenSearch benchmark is still needed and is a great idea! Thanks for pointing that out! I'll address that.

Created the linked tasks below:
- Cleanup: https://opensearch.atlassian.net/browse/MIGRATIONS-1270
- Analyze Output: https://opensearch.atlassian.net/browse/MIGRATIONS-1271

**Review by sumobrian — APPROVED** [review]

**Pull request merged**